### DISCUSSION TITLE
Styling for 'topic' directives

### DISCUSSION THREAD

**Discussion by tuukkamustonen** [pradyunsg/furo#256] [Ideas] [answered]
There seems to be no styling for the [topic](https://docutils.sourceforge.io/docs/ref/rst/directives.html#topic) element:

```rst
.. topic:: Topic Title

    Subsequent indented lines comprise
    the body of the topic, and are
    interpreted as body elements.
```

Generates:

![image](https://user-images.githubusercontent.com/94327/134632982-2c7fe3af-40ca-4657-869f-74a1f6b76b01.png)

In my previous theme, I've styled topics as somehting like:

![image](https://user-images.githubusercontent.com/94327/134633387-82055124-f298-413f-b9bd-2e193e6d89aa.png)

Simply to make them stand out a bit more. Would be neat to have something similar for Furo.

**Comment by pradyunsg** [comment]
Do you have website links, where I can see A/B comparisions of what you're seeing vs what you want to see?

Examples of how some other themes handle this would be nice as well.

It would also be helpful if you could share the ReST markup for the example; which seems distinctly different from what you've posted. :)

**Reply by tuukkamustonen** [reply]
Unfortunately no, I don't have a site up. And `topic` is not included in Furo's site (should be added to the [Kitchen Sink](https://pradyunsg.me/furo/kitchen-sink/)).

Readthedocs theme [does have it](https://sphinx-rtd-theme.readthedocs.io/en/stable/demo/demo.html#topics-sidebars-and-rubrics). It looks quite dull:

![image](https://user-images.githubusercontent.com/94327/134648380-95e11811-536a-4dda-944b-4ad91db8f84a.png)

The same can be achieved/confused with just:

```rst
**Topic title**

This is a topic.
```

...so I don't see that styling as very helpful. Styled like that, it doesn't pop up any way and the directive is just wasted.

The picture in the original post is from my customizations over readthedocs theme (but it's in internal network, no access from outside).

The given markup is for the first picture. For the second one it is:

```rst
..  topic: `Example 1 <https://example.com>`_

    *Lorem ipsum....*

..  topic: Example 2

    - Lorem ipsum.
    - Lorem ispum.
```

**Comment by cjw296** [comment]
I tend to use topics to indicate a file path and its contents. @tuukkamustonen's example above are both fine, although I'd prefer the text left-aligned rather than centered.

**Comment by pradyunsg** [comment, marked as answer]
From https://docutils.sourceforge.io/docs/ref/rst/directives.html#admonitions:

> Admonitions are specially marked "topics" that can appear anywhere an ordinary body element can. They contain arbitrary body elements. 

From https://docutils.sourceforge.io/docs/ref/rst/directives.html#topic:

> A topic is like a block quote with a title, or a self-contained section with no subsections. Use the "topic" directive to indicate a self-contained idea that is separate from the flow of the document.

---

Alrighty. They're gonna get stylized like admonitions.

**Reply by pradyunsg** [reply]
Done in e6809a2e80138c2a7f2b4f073bcec9c1d652e65a

**Reply by cjw296** [reply]
Cool, have you added an example in the kitchen sink?

**Reply by pradyunsg** [reply]
It was there already, in the paragrapharkup section. :)

**Reply by cjw296** [reply]
That looks fantastic! Icing on the cake would be a `conf.py` option to pick the icon.

In fact, really what I'm after is to have some way to pick up or configure the icon based on the class of the topic.
For example:
```
.. topic:: example.cfg
 :class: read-file
 ```
 ...would be lovely to be able to have that use a file icon in some way!
